libs/util/types/src/lib/course.ts

```diff
@@ -19,6 +19,23 @@ export type CourseCompletion = {
 	completedLessons: CompletedLessonsMap;
 };
 
+export type EnrollmentWithDetails = {
+	completedAt: string | null;
+	status: EnrollmentStatus;
+	lastProgressUpdate: string;
+	course: {
+		title: string;
+		slug: string;
+		imgUrl: string | null;
+		authors: { displayName: string }[];
+	};
+	completions: {
+		courseCompletion: {
+			completionPercentage: number;
+		};
+	};
+};
```

Comment:
benutz hier lieber type inference wenn möglich. 

wenn ich das richtig sehe, hast du gemacht, weil du die rückgabewerte aus der select methode im enrollment.router als parameter in einer komponente haben möchtest. 

exportier lieber den aufgelösten typ der select methode:

```ts
export type EnrollmentDetails = ResolvedValue<typeof getEnrollmentsOfUserWithCourseDetails>
```

Reply:
Das geht so leider nicht weil daraus eine Circular Dependency entsteht. getEnrollmentsOfUserWithCourseDetails returend EnrollmentDetails.

Reply:
ja, der rückgabewert von getEnrollmentsOfUserWithCourseDetails wird aufgelöst und als typ exportiert. ich sehe noch nicht wo der zyklus entsteht.

Reply:
Weil du sagst, die Funktion gibt x zurück und dann sagst du, x ist der Typ von der Funktion. Also:
F => X
X = typeof(X)